posthog_test.go

```diff
@@ -853,10 +853,212 @@ func TestIsFeatureEnabled(t *testing.T) {
 	}
 }
 
+func TestGetFeatureFlagPayloadWithNoPersonalApiKey(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if strings.HasPrefix(r.URL.Path, "/decide") {
+			w.Write([]byte(fixture("test-decide-v3.json")))
+		} else if !strings.HasPrefix(r.URL.Path, "/batch") {
+			t.Errorf("client called an endpoint it shouldn't have: %s", r.URL.Path)
+		}
+	}))
+	defer server.Close()
+
+	client, _ := NewWithConfig("Csyjlnlun3OzyNJAafdlv", Config{
+		Endpoint: server.URL,
+		Logger:   testLogger{t.Logf, t.Logf},
+		Callback: testCallback{
+			func(m APIMessage) {},
+			func(m APIMessage, e error) {},
+		},
+	})
+	defer client.Close()
+
+	// Test GetFeatureFlagPayload single scenario
+	payload, err := client.GetFeatureFlagPayload(FeatureFlagPayload{
+		Key:        "enabled-flag",
+		DistinctId: "test-user",
+	})
+
+	if err != nil {
+		t.Errorf("Expected no error, got: %v", err)
+	}
+
+	// Check that the flag payload is as expected (should match the value in the fixture)
+	expectedPayload := "{\"foo\": 1}"
+	if payload != expectedPayload {
+		t.Errorf("Expected flag payload %v, got: %v", expectedPayload, payload)
+	}
+
+	// should we be capturing an event here?
+	// lastEvent := client.GetLastCapturedEvent()
+	// if lastEvent == nil || lastEvent.Event != "$feature_flag_called" {
+	// 	t.Errorf("Expected a $feature_flag_called event, got: %v", lastEvent)
+	// }
+
+	// Check that the properties of the captured event match the response from /decide
+	// if lastEvent != nil {
+	// 	if lastEvent.Properties["$feature_flag"] != "beta-feature" {
+	// 		t.Errorf("Expected feature flag key 'beta-feature', got: %v", lastEvent.Properties["$feature_flag"])
+	// 	}
+	// 	if lastEvent.Properties["$feature_flag_response"] != expectedPayload {
+	// 		t.Errorf("Expected feature flag response %v, got: %v", expectedPayload, lastEvent.Properties["$feature_flag_response"])
+	// 	}
+	// }
```

Comment:
```suggestion
```

yeah, this is a reasonable callout!  I included this originally so that I could test events being captured correctly, but it doesn't make sense to include it in the SDK overall since it's one extra capture call that's not strictly necessary.  I'm fine to remove this.